### PR TITLE
Update TechDocs frontend plugin for composability system

### PR DESCRIPTION
**Pull request by adrianke77**
Signed-off-by: Chongyang Adrian, Ke <ftt.adrian.ke@grabtaxi.com>

## Hey, I just made a Pull Request!

Update Techdocs frontend plugin to composability api as first step for custom documentation landing page. 

Please refer to https://github.com/backstage/backstage/issues/4757

#### :heavy_check_mark: Checklist

- [ :heavy_check_mark: ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ NA ] Added or updated documentation
- [ NA ] Tests for new functionality and regression tests for bug fixes
- [ NA ] Screenshots attached (for UI changes)
- [ :heavy_check_mark: ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Review by emmaindal — APPROVED** [review]
lgtm reading these docs https://backstage.io/docs/plugins/composability#porting-existing-plugins (some of it has already been covered as you mentioned) but a closer eye from @backstage/maintainers would be good as well!

File: packages/app/src/components/catalog/EntityPage.tsx [hunk 1]
```diff
@@ -286,7 +286,7 @@ const ServiceEntityPage = ({ entity }: { entity: Entity }) => (
     <EntityPageLayout.Content
       path="/docs/*"
       title="Docs"
-      element={<DocsRouter entity={entity} />}
+      element={<DocsRouter />}
```
Comment (Fox32) [inline packages/app/src/components/catalog/EntityPage.tsx:289]: Based on https://backstage.io/docs/plugins/composability#porting-existing-plugins, should this be called `EntityDocsContent`?

Comment (emmaindal) [reply packages/app/src/components/catalog/EntityPage.tsx:289]: hmmm yes I believe it should use the  routable extension  https://github.com/backstage/backstage/blob/master/plugins/techdocs/src/plugin.ts#L108 instead?

Comment (Fox32) [reply packages/app/src/components/catalog/EntityPage.tsx:289]: And maybe sub routes refs also play a role here, if you have a router inside your routable component.

**Comment by Rugvip** [comment]
Hi! This is included in this PR, which migrates all of the entity pages: https://github.com/backstage/backstage/pull/4822

The changes in the TechDocs plugin here are removing deprecated pieces, which we will want to do eventually but keep in place to avoid breaking changes until we have fully migrated all plugins and given plenty of time for apps to do the same.

File: packages/app/src/components/catalog/EntityPage.tsx (hunk 1 above)
Comment (adrianke77) [reply packages/app/src/components/catalog/EntityPage.tsx:289]: @Fox32 @emmaindal it seems these changes would be part of a refactor of the catalog in the main app and outside scope of this PR?

Comment (adrianke77) [reply packages/app/src/components/catalog/EntityPage.tsx:289]: EntityTechdocsContent is the routable extension exported by the Techdocs while DocsRouter is the plain router component. It looks like to use the routable extension in the app, EntityPage.tsx has to be ported from using EntityPageLayout to EntityLayout. I think that would be under the scope of porting the main app and outside this PR's scope of porting the Techdocs frontend plugin.

**Comment by adrianke77** [comment]
> Hi! This is included in this PR, which migrates all of the entity pages: #4822
> 
> The changes in the TechDocs plugin here are removing deprecated pieces, which we will want to do eventually but keep in place to avoid breaking changes until we have fully migrated all plugins and given plenty of time for apps to do the same.

Shall I close this PR then?

**Pull request closed without merging**